enhancements/machine-config/os-images-streams.md

```diff
@@ -0,0 +1,1075 @@
+- **Gradually migrate** from one OS version to another with minimal risk
+- **Validate compatibility** of workloads with new OS releases on production
+infrastructure
+- **Run specialized OS configurations** for specific node pools without affecting
+the entire cluster
+
+### User Stories
+
+- **Specify OS stream per pool**: Set `spec.osImageStream` on a MachineConfigPool to provision nodes with a different OS version than the rest of the cluster
+- **Discover available streams**: Query the OSImageStream cluster resource to see available OS streams in the release payload
+- **Monitor stream adoption**: View `status.osImageStream` to verify when a pool has successfully adopted a new stream
+- **Backward compatibility**: Existing clusters continue working without changes; the feature is opt-in via the OSStreams feature gate
+
+### Goals
+
+- **Enable per-pool OS stream selection**: Specify different OS streams at the MachineConfigPool level for multi-OS deployments (e.g., RHCOS 9 and RHCOS 10)
+- **De-risk major OS version upgrades**: Separate platform upgrades from OS upgrades, allowing phased migration
+- **Support RHCOS 9 to RHCOS 10 transition**: Enable one-directional migration path from RHEL 9 to RHEL 10 during OpenShift 4.21-5.x timeframe. This is the primary focus for Tech Preview.
+- **Day-zero RHEL 10 deployments (stretch goal)**: While the architecture supports installing new clusters directly with RHEL 10, this is a nice-to-have for Tech Preview. If implementation complexity blocks delivery, the feature can ship without day-zero RHEL 10 support, focusing solely on RHEL 9 → RHEL 10 migration for existing clusters.
+- **API-driven stream management**: Declarative, Kubernetes-native API for stream selection
+- **Automatic stream discovery**: Populate available OS streams from release payload ImageStream metadata
+- **Backward compatibility**: Existing clusters continue working; streams are opt-in via feature gate
+- **Multi-source stream configuration**: Support CLI arguments, release ImageStream, and ConfigMap sources with defined precedence
+
+### Non-Goals
+
+- **Supporting unlimited concurrent OS streams**: While the architecture supports
+multiple OS streams, this initial implementation focuses on enabling 2-3
+concurrent streams (e.g., RHCOS 9 + RHCOS 10, or RHCOS 10 standard + minimal).
+Supporting large numbers of concurrent streams with complex version matrices is
+not a goal for the initial releases.
+
+- **HCP/Hypershift architecture support**: Supporting OS streams in
+HCP/Hypershift environments (which use NodePools instead of MachinConfigPools)
+is deferred to future work. See [Topology Considerations](#topology-considerations)
+for details.
+
+- **Automatic migration or upgrade paths**: The enhancement does not include
+automated migration logic or upgrade orchestration. Administrators must manually
+select streams for their pools.
+
+- **Bidirectional stream switching**: Only RHEL 9 → RHEL 10 migration is supported
```

Comment:
As somewhat of a general comment, I feel like part of the enhancement describes the overall goal of multi-streams, but some of the enhancement seems to want to frame this as a RHEL9->RHEL10 transition. I think we should be a bit more clear around the framing. So we would either:
1. scope the enhancement to RHEL9->RHEL10 transition as the plan for now, and explicitly call out not including other streams in other sections
2. take a phased approach and clearly call out the boundaries of each phase (starting with rhel9->10, adding more in the future)

Right now reading some of the background it sounds like we're introducing a whole variety of streams which is not the immediate plan, so I think it would be good to have some clarity for the readers.